apple/OmnivoreKit/Sources/Services/AudioSession/AudioSession.swift

```diff
@@ -0,0 +1,240 @@
+//
+//  AudioSession.swift
+//
+//
+//  Created by Jackson Harper on 8/15/22.
+//
+
+import AVFoundation
+import CryptoKit
+import Foundation
+import MediaPlayer
+import Models
+import Utils
+
+public enum AudioSessionState {
+  case stopped
+  case paused
+  case loading
+  case playing
+}
+
+// Our observable object class
+public class AudioSession: ObservableObject {
+  @Published public var state: AudioSessionState = .stopped
+  @Published public var item: LinkedItem?
+
+  let appEnvironment: AppEnvironment
+  let networker: Networker
+
+  var timer: Timer?
+  var player: AVAudioPlayer?
+
+  public init(appEnvironment: AppEnvironment, networker: Networker) {
+    self.appEnvironment = appEnvironment
+    self.networker = networker
+  }
+
+  public func play(item: LinkedItem) {
+    // Stop any existing session
+    stop()
+
+    self.item = item
+    startAudio()
+  }
+
+  public func stop() {
+    player?.stop()
+    clearNowPlayingInfo()
+    timer = nil
+    player = nil
+    item = nil
+    state = .stopped
+  }
+
+  public func isLoadingItem(item: LinkedItem) -> Bool {
+    state == .loading && self.item == item
+  }
+
+  public func isPlayingItem(item: LinkedItem) -> Bool {
+    state == .playing && self.item == item
+  }
+
+  public func startAudio() {
+    state = .loading
+
+    let pageId = item!.unwrappedID
+
+    Task {
+      do {
+        try await downloadAudioFile(pageId: pageId)
+      } catch {
+        print("FAILED TO DOWNLOAD AUDIO URL")
+        print(error.localizedDescription)
```

Comment:
Right now we intentionally fall through here, so the error gets detected and handled on the main thread.